### PR TITLE
Basic Sign Up / Sign In

### PR DESCRIPTION
**Pull request by Jma353**
UI for basic sign up and sign in.  Essentially just hooking up the UI to endpoints / establishing a basic redux flow.  Addresses #38 and #39

### CODE REVIEW

**Comment by Jma353** [comment]
Added @kq-li

File: src/client/js/components/SignIn.react.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          onClick={this.handleClick}
+        />
+        {this.state.error && (<div className='error'>{this.state.error}</div>)}
+      </div>
+    );
+  }
+}
+
+const acts = (dispatch: Function) => {
+  const signIn = (
```
Comment (kq-li) [inline src/client/js/components/SignIn.react.js]: Shorthand? `const signIn = (signInAttempt: SignInAttempt), errorHandler: Function) => dispatch(actions.signIn(signInAttempt, errorHandler));` Not really important, just something I like to do

Comment (kq-li) [reply src/client/js/components/SignIn.react.js]: Also, https://flow.org/en/docs/types/functions/#toc-function-type?

File: src/client/js/components/actions/AuthActions.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// @flow
+import type {
+  Session,
+  SignUpCredentials,
+  SignInAttempt
+} from '../common/types';
+import AuthActionTypes from './AuthActionTypes';
+import HTTPRequest from '../common/HTTPRequest';
+
+const setSession = (session: Session) => {
```
Comment (kq-li) [inline src/client/js/components/actions/AuthActions.js]: Shorthand arrow function works for these too

File: src/client/js/components/common/types.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+// Received from the server
+export type SessionData = {
+  session_token: string,
+  session_expiration: string,
+  update_token: string
+}
+
+// Stored in memory / as a cookie
+export type Session = {
```
Comment (kq-li) [inline src/client/js/components/common/types.js]: Change from snake to camel case?

**Pull request merged**